orthoinference/src/main/java/org/reactome/orthoinference/InferEvents.java

```diff
@@ -0,0 +1,307 @@
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Properties;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+import org.gk.schema.GKSchemaAttribute;
+import org.gk.schema.SchemaClass;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.reactome.release.common.database.InstanceEditUtils;
+
+/**
+ *
+ * @author jcook
+ * 
+ * The Java version of infer_events.pl -- The gist of this module is that it looks at all existing Human ReactionlikeEvent (RlE) instances (mostly Reactions and BlackBoxEvents) in the Test_Reactome database,
+ * and attempts to computationally infer them in each of Reactome's model organisms. Each RlE is broken down into its primary components (input, output, catalyst, and regulator), which are themselves broken
+ * into their PhysicalEntity subunits. The homology data used for the inference process currently comes from Ensembl Compara and is generated during the 'Orthopairs' step of the Reactome release process.
+ * After all inference attempts for each RlE has been completed in an organism, the pathways that contain the reactions are filled with these newly inferred ones. 
+ * 
+ *
+ */
+
+// To Reviewers: I recommend following the code sequentially, rather then looking at each class separately.
+public class InferEvents
+{
+	static MySQLAdaptor dbAdaptor = null;
+	private static GKInstance instanceEdit;
+	private static GKInstance speciesInst;
+	private static HashMap<GKInstance,GKInstance> manualEventToNonHumanSource = new HashMap<GKInstance,GKInstance>();
+	private static ArrayList<GKInstance> manualHumanEvents = new ArrayList<GKInstance>();
+
+	@SuppressWarnings("unchecked")
+	public static void eventInferrer(Properties props, String pathToConfig, String pathToSpeciesConfig) throws Exception
+	{
+		// Set up DB adaptor using config.properties file
+		String speciesToInferFromLong = "Homo sapiens";
+		String username = props.getProperty("username");
+		String password = props.getProperty("password");
+		String database = props.getProperty("database");
+		String host = props.getProperty("host");
+		int port = Integer.valueOf(props.getProperty("port"));
+		
+		dbAdaptor = new MySQLAdaptor(host, database, username, password, port);
+		// Statically store the adaptor variable in each class
+		InferReaction.setAdaptor(dbAdaptor);
+		SkipTests.setAdaptor(dbAdaptor);
+		GenerateInstance.setAdaptor(dbAdaptor);
+		OrthologousEntity.setAdaptor(dbAdaptor);
+		InferEWAS.setAdaptor(dbAdaptor);
+		UpdateHumanEvents.setAdaptor(dbAdaptor);
```

Comment:
Maybe these could all be set together in one method call like with 'resetVariables'

Reply:
Yep that would clean this part of the code a good bit